java/pom.xml

```diff
@@ -58,9 +58,77 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-shade-plugin</artifactId>
-        <version>2.3</version>
+        <version>3.1.1</version>
         <configuration>
-          <createDependencyReducedPom>false</createDependencyReducedPom>
+          
+          <!-- The following settings can be used to create a JAR which is -->
+          <!-- smaller in size, this can be used in the event that the JAR -->
+          <!-- is too big for Amazon Lambda. -->
+          <!-- Note that the setup here works for this example project and -->
+          <!-- it may or may not work for other projects if they add more -->
+          <!-- dependencies. Running the resulting JAR may work on some -->
+          <!-- invocations or it may fail on other invocations depending -->
+          <!-- on the classes available that can be initialized. -->
+          <!-- If an invocation fails with NoClassDefFoundError or -->
+          <!-- ExceptionInInitializationError then a class which is -->
+          <!-- required has not been included. Going through the project -->
+          <!-- dependency tree you should be able to determine the project -->
+          <!-- that must be included. -->
+          <!-- Or, this entire section can just not be included in which -->
+          <!-- case any issues that might arrive will not occur. -->
+          <createDependencyReducedPom>true</createDependencyReducedPom>
+          <minimizeJar>true</minimizeJar>
+          <filters>
+             <!-- These are needed by IOpipe and the examples to function. -->
+             <filter>
+               <artifact>com.iopipe:iopipe</artifact>
+               <includes>**</includes>
+             </filter>
+             <filter>
+               <artifact>org.glassfish:javax.json</artifact>
+               <includes>**</includes>
+             </filter>
+             <filter>
+               <artifact>org.apache.logging.log4j:*</artifact>
+               <includes>**</includes>
+             </filter>
+             <filter>
+               <artifact>com.amazonaws:aws-lambda-java-core</artifact>
+               <excludes>**</excludes>
+             </filter>
+             <filter>
+               <artifact>com.amazonaws:aws-lambda-java-events</artifact>
+               <includes>**</includes>
+             </filter>
+             <filter>
+               <artifact>com.amazonaws:aws-java-sdk-kinesis</artifact>
+               <includes>**</includes>
+             </filter>
+             
+             <!-- These are not needed by AWS as they are pre-included -->
```

Comment:
If they are not needed, why are they included?

Reply:
Oh! This is a list of dependencies which are explicitly excluded otherwise they would be implicitly included by dependencies since they are actually used by the examples.